python/ray/serve/controller.py

```diff
@@ -484,6 +452,28 @@ class ServeController:
           requires all implementations here to be idempotent.
     """
 
+    class StateNotifier:
+        def __init__(self, controller: "ServeController"):
+            self.controller = controller
+            self.long_pull_host = controller.long_pull_host
+            self.long_pull_host.notify_on_changed("traffic_policies", dict())
+            self.long_pull_host.notify_on_changed("worker_handles", dict())
+            self.long_pull_host.notify_on_changed("backend_configs", dict())
+
+        def on_replica_handles_changed(self):
+            self.long_pull_host.notify_on_changed(
+                "worker_handles", self.controller.get_all_worker_handles())
+
+        def on_traffic_policies_changed(self):
+            self.long_pull_host.notify_on_changed(
+                "traffic_policies",
+                self.controller.configuration_store.traffic_policies)
+
+        def on_backend_configs_change(self):
```

Comment:
Does this mean that all backends in the system listen for changes to all other backends' configs? That seems potentially problematic at scale. Can we leave a comment at a minimum saying that we should extend this to allow subscription for a specific key?

Reply:
If that's not the case, then the interface isn't obvious on its own, so we should make sure to document that clearly